integration_test/third_party_apps_test.go

```diff
@@ -510,14 +478,40 @@ const (
 	SAPHANAApp      = "saphana"
 )
 
+// incompatibleOperatingSystem looks at the supported_operating_systems field
+// of metadata.yaml for this app and returns a nonempty skip reason if it
+// thinks this app doesn't support the given platform.
+// supported_operating_systems should only contain "linux", "windows", or
+// "linux_and_windows".
+func incompatibleOperatingSystem(t *testing.T, testCase test) string {
+	supported := testCase.metadata.SupportedOperatingSystems
+	if supported == "linux_and_windows" {
+		return "" // This app supports both Linux and Windows.
+	}
+	if supported == "windows" {
+		if !gce.IsWindows(testCase.platform) {
+			return fmt.Sprintf("Skipping test for platform %v because app %v only supports Windows.", testCase.platform, testCase.app)
+		}
+		return "" // We are testing Windows and this app supports Windows.
+	}
+	if supported == "linux" {
+		if gce.IsWindows(testCase.platform) {
+			return fmt.Sprintf("Skipping test for platform %v because app %v only supports Linux.", testCase.platform, testCase.app)
+		}
+		return "" // We are testing Linux and this app supports Linux.
+	}
+	t.Fatalf("Unrecognized value %q for supported_operating_systems.", supported)
```

Comment:
This case is probably not necessary given the validation rules… You could make this a `panic` instead…

Reply:
obsolete